src/snowflake/snowpark/modin/plugin/_internal/indexing_utils.py

```diff
@@ -2068,6 +2072,76 @@ def set_frame_2d_labels(
     # map from item's data column label to its position in the joined frame or item_column_values
     item_data_col_label_to_pos_map: dict[Hashable, int] = {}
     if item_is_frame:
+        from snowflake.snowpark.modin.plugin.compiler.snowflake_query_compiler import (
```

Comment:
why not move them into `_set_2d_labels_helper_for_frame_item` since item is a frame.

Reply:
This codepath also modifies index, which is used later on in the post-processing steps of the current function, so it makes sense to perform the computation here imo.

Reply:
can you wrap them into a helper function and we can clearly see what's the input and output are?

Reply:
sure!